sdk/core/core-rest-pipeline/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -170,7 +170,8 @@ export function bearerTokenAuthenticationPolicy(
      * - Retrieve a token with the challenge information, then re-send the request.
      */
     async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
-      if (!request.url.toLowerCase().startsWith("https://")) {
+      const lowerCaseUrl = request.url.toLowerCase();
+      if (!lowerCaseUrl.startsWith("https://") && !lowerCaseUrl.startsWith("http://localhost")) {
```

Comment:
Scenario: attacker registers the domain `localhostisacoolplacetohackpeople.com` and we trivially match against it

Also it feels a bit arbitrary to allow `localhost` but not `127.0.0.1` or other loopback addresses.

Reply:
Also .NET seems to have the same restriction: https://github.com/Azure/azure-sdk-for-net/blob/99841c9d4f42e3fdc538e9b2fe248227146e9d0f/sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs#L117

Reply:
Same with Java: https://github.com/Azure/azure-sdk-for-java/blob/4770e3cc599f0361f85f9b097dcff63cd75a94f0/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/BearerTokenAuthenticationPolicy.java#L96

Reply:
Python appears to be the only language that currently lets you disable the check: https://github.com/Azure/azure-sdk-for-python/blob/d550bb3bc1d404081ecc7e735a1ee947d63aedab/sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py#L49

Not sure if @lmazuel knows why Python decided to deviate here?